### ISSUE TITLE
modsec module does not work

### ISSUE THREAD

**Issue by aeimer** [bitnami/bitnami-docker-nginx-ingress-controller#10]
**Description**

When I activate the `enable-modsecurity` option the new config cannot be loaded

**Steps to reproduce the issue:**

1. fire up the ingress nginx helm chart `bitnami/nginx-ingress-controller:7.6.10`
2. update the config to include

```yaml
config:
  enable-modsecurity: 'true' # Buggy!
  enable-owasp-modsecurity-crs: 'true' # Use the default rules of modsec => It even fails without this flag.
```

**Describe the results you received:**

```log
I0618 12:08:49.471908 1 controller.go:146] "Configuration changes detected, backend reload required"
E0618 12:08:49.848267 1 controller.go:158] Unexpected failure reloading the backend:

-------------------------------------------------------------------------------
Error: exit status 1
2021/06/18 12:08:49 [emerg] 239#239: "modsecurity_rules_file" directive Failed to open file: /var/log/modsec_audit.log in /tmp/nginx-cfg018561431:118
nginx: [emerg] "modsecurity_rules_file" directive Failed to open file: /var/log/modsec_audit.log in /tmp/nginx-cfg018561431:118
nginx: configuration file /tmp/nginx-cfg018561431 test failed

```

**Describe the results you expected:**

Id expect to see the modsec logs in the pod log.

**Additional information you deem important (e.g. issue happens only occasionally):**

**Version**

- Output of `docker version`:

```
We run on k8s 1.20
```

- Output of `docker info`:

```
(paste your output here)
```

- Output of `docker-compose version` (if applicable):

```
(paste your output here)
```

**Additional environment details (AWS, VirtualBox, Docker for MAC, physical, etc.):**

**Comment by pablogalegoc** [comment]
Hi @aeimer! I've reproduced the issue and will keep you updated on what I find

**Comment by pablogalegoc** [comment]
Hi again @aeimer!

I will create an internal task to address this bug, thanks for catching this!

**Comment by pablogalegoc** [comment]
The [0.47.0-debian-10-r12 release](https://github.com/bitnami/bitnami-docker-nginx-ingress-controller/releases/tag/0.47.0-debian-10-r12) of the image should fix this issue and the latest version of the `nginx-ingress-controller` chart is [already using it](https://github.com/bitnami/charts/blob/bfed122a747cacb6958fc5d1b460ecb8b48b88cd/bitnami/nginx-ingress-controller/values.yaml#L16). Let me know if you run into any further issues with this!

**Comment by aeimer** [comment]
@pablogalegoc Yes this looks really good! The pod starts and `cat /var/log/modsec_audit.log` shows the correct logs.

Is there any best practice how to retrieve and process the log as this file is not redirected to `stdout`? I mean it's good to not have it there I guess, but it makes things more complicated on the other hand?

**Comment by pablogalegoc** [comment]
@aeimer I don't know best practices in that respect, sorry. If redirecting the to `stdout` could serve for your use case you can always override the entrypoint with a `tail -f` or similar, but you probably know that.

**Comment by aeimer** [comment]
@pablogalegoc I see. I guess I need to setup the logging solution to also checking that file or so...

Let's close the ticket as the problem does not occur anymore.

Thanks for the quick fix, best regards!
